modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/category/facet/portlet/CategoryFacetPortlet.java

```diff
@@ -113,26 +112,26 @@ private AssetCategoriesSearchFacetDisplayContext _buildDisplayContext(
 		PortletSharedSearchResponse portletSharedSearchResponse,
 		RenderRequest renderRequest) {
 
-		Facet facet = portletSharedSearchResponse.getFacet(
-			_getAggregationName(renderRequest));
+		AssetCategoriesSearchFacetDisplayContextBuilder
+			assetCategoriesSearchFacetDisplayContextBuilder =
+				new AssetCategoriesSearchFacetDisplayContextBuilder(
+					renderRequest);
+
+		assetCategoriesSearchFacetDisplayContextBuilder.setFacet(
```

Comment:
@joshuacords @felipelorenz i think setFacet() should still come after setDisplayStyle()

the reason it comes after the declaration of userSearchFacetDisplayContextBuilder in
https://github.com/gustavolimav/liferay-portal/pull/39/commits/ed6864294ea6f1c8f5ca9c4159da32cbef364d40#diff-7a063f967f58021b2ffd4ac20198ad1c56cd20cbaf4b4a64abf0f78a7df68dc4R112
is because it was the first method called on userSearchFacetDisplayContextBuilder

i think the general pattern should be this

1) DisplayContextBuilder declared first
2) DisplayContextBuilder methods called in alphabetical order
3) if a variable is only used once after being declared, just "inline" it inside the setter call, ie what we did with setFacet 
4) if a variable is declared and used more than once, add line breaks between calls to the DisplayContextBuilder setters and declare it there, immediately before its used for the first time, ie categoryFacetPortletPreferences should go between setAssetVocabularyLocalService and setDisplayStyle
5) there should be a line break after the "last" usage of a variable that was used more than once, ie categoryFacetPortletPreferences (sourceformatter should enforce this)

the code in the method is really scattered... towards the bottom im seeing calls like setExcludedGroupId and setAssetCategoryPermissionChecker which should come way earlier according to the general pattern..... the only thing i can think of is that the original code is so old that it didnt have to conform to these standards when it was written, or was simply missed by a review

Reply:
i think this branch/pr was actually closer to what he wanted https://github.com/brianchandotcom/liferay-portal/pull/124650

Reply:
it should potentially look like this https://github.com/BryanEngler/liferay-portal/commit/e934cf5029e3433ddff2daac4678ed5df786378b#diff-e05828c8e903b29f2a4df3d239a653c6cebcf27516fcf55951472bc9ce769214